dali/python/nvidia/dali/plugin/mxnet.py

```diff
@@ -423,3 +459,224 @@ def __init__(self,
+            auto_reset)
+
+        # Build all pipelines
+        for p in self._pipes:
+            with p._check_api_type_scope(types.PipelineAPIType.ITERATOR):
+                p.build()
+        self._data_batches = [None for i in range(self._num_gpus)]
+        self._counter = 0
+        self._output_tags = {DALIGluonIterator.DENSE_TAG, DALIGluonIterator.SPARSE_TAG}
+        assert output_types is None or set(output_types) <= self._output_tags, \
+            "Only DENSE_TAG and SPARSE_TAG are allowed"
+
+        self._outputs_types = output_types
+
+        # We need data about the batches (like shape information),
+        # so we need to run a single batch as part of setup to get that info
+        for p in self._pipes:
+            with p._check_api_type_scope(types.PipelineAPIType.ITERATOR):
+                p.schedule_run()
+        self._first_batch = None
+        self._first_batch = self.next()
+
+
+    def __next__(self):
+        if self._first_batch is not None:
+            batch = self._first_batch
+            self._first_batch = None
+            return batch
+        if self._counter >= self._size and self._size > 0:
+            if self._auto_reset:
+                self.reset()
+            raise StopIteration
+        # Gather outputs
+        dali_outputs = []
+        for p in self._pipes:
+            with p._check_api_type_scope(types.PipelineAPIType.ITERATOR):
+                dali_outputs.append(p.share_outputs())
+        for i in range(self._num_gpus):
+            output_elements = []
+            shapes = []
+            for j, out in enumerate(dali_outputs[i]):
+                if self._outputs_types is None or self._outputs_types[j] == DALIGluonIterator.DENSE_TAG:
+                    output_elements.append(out.as_tensor())
+                    shapes.append(output_elements[-1].shape())
+                else:
+                    output_elements.append([out[sample_idx] for sample_idx in range(self.batch_size)])
+                    s = [t.shape() for t in output_elements[-1]]
+                    shapes.append(s)
+
+            if self._data_batches[i] is None:
+                self._data_batches[i] = self._create_data_batch(output_elements, shapes, self._pipes[i].device_id)
+
+            batch = self._data_batches[i]
+            # Copy data from DALI Tensors to MXNet NDArrays
+            for j, output_el in enumerate(output_elements):
+                if self._outputs_types is None or self._outputs_types[j] == DALIGluonIterator.DENSE_TAG:
+                    ndarray = batch[j].resize(shapes[j])
+                    feed_ndarray(output_el, ndarray)
+                else:
+                    for sample_idx in range(self.batch_size):
+                        ndarray = batch[j][sample_idx].resize(shapes[j][sample_idx])
+                        feed_ndarray(output_el[sample_idx], ndarray)
+
+        batches = [[([sample.array for sample in output_el] if isinstance(output_el,list) else output_el.array)
```

Comment:
Shouldn't we return the `view` instead of `array`? It's the former that is reshaped properly.